flank-scripts/src/main/kotlin/flank/scripts/integration/Extensions.kt

```diff
@@ -0,0 +1,66 @@
+package flank.scripts.integration
+
+import com.github.kittinunf.result.getOrNull
+import com.github.kittinunf.result.onError
+import flank.scripts.github.objects.GitHubCreateIssueCommentRequest
+import flank.scripts.github.objects.GitHubCreateIssueRequest
+import flank.scripts.github.objects.GitHubCreateIssueResponse
+import flank.scripts.github.objects.GitHubUpdateIssueRequest
+import flank.scripts.github.objects.IssueState
+import flank.scripts.github.patchIssue
+import flank.scripts.github.postNewIssue
+import flank.scripts.github.postNewIssueComment
+import flank.scripts.utils.toJson
+import kotlinx.coroutines.coroutineScope
+import kotlin.system.exitProcess
+
+internal suspend fun IntegrationContext.createNewIssue() = coroutineScope {
+    println("** Creating new issue")
+    val issuePayload = GitHubCreateIssueRequest(
+        title = "Full Suite integration tests failed on master",
+        body = "### Integration Test failed on master",
+        labels = listOf("IT_Failed", "bug")
+    )
+    println(issuePayload.toJson())
+    val issue = postNewIssue(token, issuePayload)
+        .onError {
+            println(it.message)
+        }
+        .getOrNull()
+    if (issue == null) {
+        println("** Issue created but empty response returned. Terminating.")
+        exitProcess(-1)
+    }
+    logIssueCreated(issue)
+    copy(openedIssue = issue.number)
```

Comment:
I would split it to
function 1:
```kotlin
    println("** Creating new issue")
    val issuePayload = GitHubCreateIssueRequest(
        title = "Full Suite integration tests failed on master",
        body = "### Integration Test failed on master",
        labels = listOf("IT_Failed", "bug")
    )
    println(issuePayload.toJson())
```
function 2
```kotlin
val issue = postNewIssue(token, issuePayload)
        .onError {
            println(it.message)
        }
        .getOrNull()
    if (issue == null) {
        println("** Issue created but empty response returned. Terminating.")
        exitProcess(-1)
    }
```
function 3
```kotlin
    logIssueCreated(issue)
    copy(openedIssue = issue.number)
```

Reply:
In a bit different way but refactored :)